### PR TITLE
user profile

### PR DESCRIPTION
**Pull request by moraw9**
# tasks
- [x] create and secure a link to your profile
- [x] create template for user profile with correct data
- [x] create option to update user's data
- [x] fix if user changes the email and password simultaneously
- [x] maybe prompt is not good solution for entering a password - change it 
- [x] user name must change after update without page reload
- [x] add input to upload user photo and convert it into Base64

### CODE REVIEW

File: app/components/log-in/template.hbs [hunk 1]
```diff
@@ -20,7 +20,10 @@
           {{#if this.isLogInForm}}
             <LogIn::logInForm />
           {{else}}
-              <LogIn::registerForm @model={{this.model}}/>
+              <div class="header content-frame bottom">
+                <h1>Sign up</h1>
+              </div>
+              <UserForm @setDataToUpdate={{fn this.setDataToUpdate}} @changeset={{this.changeset}}/>
```
Comment (wozny1989) [inline app/components/log-in/template.hbs]: Indents ◀️

File: app/components/user-form/field/template.hbs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+  <div class="form-group">
+    <label for={{@name}} class="label">{{@label}}</label>
+    <input type={{@type}} class="form-control" placeholder={{get @placeholder @name}} id={{@name}} name={{@name}} onchange={{@onChange}} disabled={{@disabled}} >
+    {{#if  (get @changesetError @name) }}
+      <p class="text-danger">{{get @changesetError (concat @name ".validation") }}</p>
+    {{/if}}
+  </div>
```
Comment (wozny1989) [inline app/components/user-form/field/template.hbs:7]: Nice! 👏

File: app/components/user-form/template.hbs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  />
+  <UserForm::Field
+    @label='Surname'
+    @type='text'
+    @name='surname'
+    @onChange={{this.setValue}}
+    @changesetError={{@changeset.error}}
+    @placeholder={{@userData}}
+  />
+    <UserForm::Field
```
Comment (wozny1989) [inline app/components/user-form/template.hbs]: Generally looks awesome! ❤️  Just fix components indent
```suggestion
  <UserForm::Field
```

File: app/router.js [hunk 4]
```diff
@@ -7,5 +7,6 @@ export default class Router extends EmberRouter {
 Router.map(function () {
   this.route('idea-details', { path: '/ideas/:id' });
   this.route('log-in', { path: '/LogIn' });
-  this.route('login');
+  // this.route('log-in');
```
Comment (wozny1989) [inline app/router.js]: ✂️

File: app/components/user-form/component.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  @action
+  cancel() {
+    this.args.changeset.rollback();
+    let inputs = document.querySelectorAll('input');
+    inputs.forEach((input) => (input.value = ''));
+    if (typeOf(this.args.toggleState) === 'function') {
+      this.args.toggleState();
+    }
+  }
+  @action
```
Comment (wozny1989) [inline app/components/user-form/component.js]: In my personal suggestion, good approach is adding empty line between functions. In bigger components you'll see a wall of code, wich is little not readable.

**Review by wozny1989 — CHANGES_REQUESTED** [review]

File: app/components/user-form/field/template.hbs (hunk 2 above)
Comment (moraw9) [reply app/components/user-form/field/template.hbs:7]: <3

**Pull request merged**